applications/GeoMechanicsApplication/custom_elements/small_strain_U_Pw_diff_order_element.cpp

```diff
@@ -850,13 +850,11 @@ void SmallStrainUPwDiffOrderElement::CalculateOnIntegrationPoints(const Variable
 {
     KRATOS_TRY
 
-    const unsigned int& IntegrationPointsNumber =
+    const auto number_of_integration_points =
         GetGeometry().IntegrationPointsNumber(this->GetIntegrationMethod());
 
-    if (rValues.size() != IntegrationPointsNumber) {
-        rValues.resize(IntegrationPointsNumber);
-    }
-    for (unsigned int i = 0; i < IntegrationPointsNumber; ++i) {
+    rValues.resize(number_of_integration_points);
+    for (auto i = SizeType{0}; i < number_of_integration_points; ++i) {
```

Comment:
Could be a std::transform 😄 , but leave it up to you if you want to do it, or just merge these improvements (since we're going to tackle the rest later anyway)

Reply:
Well, in this case that is unfortunately not so clean, since the value we're trying to retrieve is also returned through the parameter list.  Bleeeh! (In addition to returning the value through the function's return value.) Since inside the lambda expression (which you need to pass to `std::transform`) you don't have access to the object where the transformed value will be stored, it means that you would need to create a dummy variable, just in order to call `GetValue`. In my opinion, that is not a clean solution (since you probably need a comment to explain what is happening there). So I'd rather leave it the way it is now. I would be in favor of changing it to `std::transform` if the function signature of `GetValue` would be improved (i.e. don't return the value through the parameter list, but only through its return value).

Reply:
No worries :-)